### PR TITLE
feat: add user agent support

### PR DESCRIPTION
**Pull request by bsatrom**

### CODE REVIEW

File: notecard/notecard.py [hunk 1]
```diff
@@ -155,16 +156,49 @@ class Notecard:
...
         self.Reset()
 
+        if sys.implementation.name == 'cpython':
+            self._user_agent['os_family'] = os.name
+        else:
+            self._user_agent['os_family'] = os.uname().machine
+
+    def _preprocessReq(self, req):
+        """Inspect the request for hub.set and add the User Agent."""
+        if 'hub.set' in req.values():
```
Comment (zfields) [inline notecard/notecard.py:178]: Is there a more computationally efficient way to look for `hub.set`, like `if (exists req['hub.set'])`?

File: notecard/notecard.py [hunk 2]
```diff
@@ -155,16 +156,49 @@ class Notecard:
...
+            # Merge the User Agent to send along with the hub.set request.
+            new_req = req.copy()
+            new_req.update(self.GetUserAgent())
+            req = new_req
```
Comment (zfields) [inline notecard/notecard.py:182]: I don't understand Python well enough to understand this extra copy of the structure.

**Review by zfields — APPROVED** [review]
Approved with comments

File: notecard/notecard.py (hunk 1 above)
Comment (bsatrom) [reply notecard/notecard.py:178]: In Python, this is the best way to search a dictionary for values. It's quite efficient, actually, and a far better option that looping over the collection.

File: notecard/notecard.py (hunk 2 above)
Comment (bsatrom) [reply notecard/notecard.py:182]: There is a better way to do this in Python 3.5+ using spread notation, but that is not supported in CircuitPython and MicroPython, hence this 3 step approach to be cross-platform.

**Pull request merged**